src/main/java/io/druid/segment/loading/DataSegmentPullerURI.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2015 Metamarkets Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.segment.loading;
+
+import com.google.common.base.Predicate;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+
+/**
+ * A DataSegmentPullerURI extends the normal DataSegmentPuller to allow URI based handlings
+ */
+public interface DataSegmentPullerURI extends DataSegmentPuller
+{
+  /**
+   * Create a new InputStream based on the URI
+   * @param uri The URI to open an Input Stream to
+   * @return A new InputStream which streams the URI in question
+   * @throws IOException
+   */
+  public InputStream getUriInputStream(URI uri) throws IOException;
+  /**
+   * Return the `Long` timestamp of when a URI was last modified
+   * @param uri The URI to check
+   * @return The time in ms since 1970-01-01Z when the segment was last modified
+   * @throws SegmentLoadingException on error
+   */
+  public Long getLastModified(URI uri) throws IOException;
```

Comment:
I typically use objects (unless I'm using an array) because they are infinitely more useful and I've not yet seen proof where simple `Long` vs `long` passing causes an actual performance hit.